src/fork_gc.c

```diff
@@ -1146,18 +1103,10 @@ static int FGC_fork(ForkGC *gc, RedisModuleCtx *ctx) {
 
 static int periodicCb(RedisModuleCtx *ctx, void *privdata) {
   ForkGC *gc = privdata;
-  if (gc->deleting) {
-    return 0;
-  }
-  if (gc->deletedDocsFromLastRun < RSGlobalConfig.forkGcCleanThreshold) {
-    return 1;
-  }
-
-  int gcrv = 1;
 
   // Check if RDB is loading - not needed after the first time we find out that rdb is not
   // reloading
-  if (gc->rdbPossiblyLoading && !gc->sp) {
+  if (gc->rdbPossiblyLoading && !gc->index.rm) {
```

Comment:
Can this even happened? `gc->index.rm`

Reply:
I don't think so. How could it happen before?